sdk/src/main/java/com/hcaptcha/sdk/HCaptchaHeadlessWebView.java

```diff
@@ -46,22 +46,17 @@ final class HCaptchaHeadlessWebView implements IHCaptchaVerifier {
     public void startVerification(@NonNull FragmentActivity activity) {
         if (webViewLoaded) {
             // Safe to execute
-            resetAndExecute();
+            webViewHelper.resetAndExecute();
         } else {
             shouldExecuteOnLoad = true;
         }
     }
 
-    private void resetAndExecute() {
-        webViewHelper.getWebView().loadUrl("javascript:resetAndExecute();");
-    }
-
     @Override
-    public void onFailure(final HCaptchaException exception) {
-        final boolean silentRetry = webViewHelper.getConfig().getResetOnTimeout()
-                && exception.getHCaptchaError() == HCaptchaError.SESSION_TIMEOUT;
+    public void onFailure(@NonNull final HCaptchaException exception) {
+        final boolean silentRetry = webViewHelper.shouldRetry(exception);
         if (silentRetry) {
-            resetAndExecute();
+            webViewHelper.resetAndExecute();
         } else {
             listener.onFailure(exception);
         }
```

Comment:
This block looks like the exact duplicate of https://github.com/hCaptcha/hcaptcha-android-sdk/pull/87/files#diff-d0637e553a517f03703d358d658b90ab0a930bb4c81793a93520fc074765a38dR205-R209

Previously there was `HCaptchaWebViewHelper.retryIfRequested` method, the name is probably can be better but it kept this code in a single place

Reply:
This very simple code is not worth abstracting. The Headless and Dialog do not necessarily have the same flow.